src/stdlib_experimental_io.f90

```diff
@@ -1,8 +1,15 @@
 module stdlib_experimental_io
 use iso_fortran_env, only: sp=>real32, dp=>real64, qp=>real128
+use stdlib_experimental_error, only: error_stop
+use stdlib_experimental_optval, only: optval
 implicit none
 private
-public :: loadtxt, savetxt
+! Public API
+public :: loadtxt, savetxt, open
+
+! Private API that is exposed so that we can test it in tests
+public :: parse_mode
```

Comment:
Not sure if it is possible with CMake, but would it be possible to use some preprocessing (cpp?) for setting `parse_mode` public only for the tests?

Reply:
We could. See also #75.